src/reasoning-layer-lib/connectors-reasoning-layer.xsl

```diff
@@ -81,96 +63,132 @@
             <xsl:with-param name="generalisation" select="."/>
         </xsl:call-template>
     </xsl:template>
-
+    
+    
+    <xd:doc>
+        <xd:desc>Applying rule 12,13 and 20 to connectors with distinct names</xd:desc>
+    </xd:doc>
+    <xsl:template name="distinctConnectorsNamesInReasoningLayer">
+        <xsl:variable name="root" select="root()"/>
+        <xsl:variable name="distinctNames" select="f:getDistinctConnectorsNames($root)"/>
+        <xsl:for-each select="$distinctNames">
+            <xsl:call-template name="connectorDomain">
+                <xsl:with-param name="connectorName" select="."/>
+                <xsl:with-param name="root" select="$root"/>
+            </xsl:call-template>
+            <xsl:call-template name="connectorRange">
+                <xsl:with-param name="connectorName" select="."/>
+                <xsl:with-param name="root" select="$root"/>
+            </xsl:call-template>
+            <xsl:call-template name="connectorInverse">
+                <xsl:with-param name="connectorName" select="."/>
+                <xsl:with-param name="root" select="$root"/>
+            </xsl:call-template>
+        </xsl:for-each>
+    </xsl:template>
 
 
     <xd:doc>
         <xd:desc>Rule 12 (Association source in reasnoning layer) .Specify object property domain
             for the target end of the association.</xd:desc>
-        <xd:param name="connector"/>
+        <xd:param name="connectorName"/>
+        <xd:param name="root"/>
     </xd:doc>
 
-    <xsl:template name="connectorSource">
-        <xsl:param name="connector"/>
-        <xsl:variable name="sourceClassURI"
-            select="f:buildURIfromLexicalQName($connector/source/model/@name, fn:true())"/>
-        <xsl:variable name="sourceRole"
-            select="
-                if (boolean($connector/source/role/@name)) then
-                    f:lexicalQNameToWords($connector/source/role/@name)
-                else
-                    concat($mockUnknownPrefix, ':', $mockUnnamedElement)"/>
-        <xsl:variable name="sourceRoleURI"
-            select="f:buildURIfromLexicalQName($sourceRole, fn:false())"/>
-        <xsl:variable name="targetClassURI"
-            select="f:buildURIfromLexicalQName($connector/target/model/@name, fn:true())"/>
-        <xsl:variable name="targetRole"
-            select="
-                if (boolean($connector/target/role/@name)) then
-                    f:lexicalQNameToWords($connector/target/role/@name)
-                else
-                    concat($mockUnknownPrefix, ':', $mockUnnamedElement)"/>
+    <xsl:template name="connectorDomain">
```

Comment:
Nice that we have switched from the UML terminology "source/target" to OWL terminology "domain/range"